adapters/smoot/smoot.go

```diff
@@ -0,0 +1,153 @@
+	PlacementID string `json:"placementId,omitempty"`
+	EndpointID  string `json:"endpointId,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+	for _, imp := range request.Imp {
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		var bidderExt adapters.ExtImpBidder
+		var smootExt openrtb_ext.ImpExtSmoot
+
+		if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &smootExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		impExt := reqBodyExt{SmootBidderExt: reqBodyExtBidder{}}
+
+		if smootExt.PlacementID != "" {
+			impExt.SmootBidderExt.PlacementID = smootExt.PlacementID
+			impExt.SmootBidderExt.Type = "publisher"
+		} else {
+			impExt.SmootBidderExt.EndpointID = smootExt.EndpointID
+			impExt.SmootBidderExt.Type = "network"
+		}
+
+		finalyImpExt, err := json.Marshal(impExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		reqCopy.Imp[0].Ext = finalyImpExt
+
+		adapterReq, err := a.makeRequest(&reqCopy)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		adapterRequests = append(adapterRequests, adapterReq)
+	}
+
+	return adapterRequests, errs
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, error) {
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	return &adapters.RequestData{
+		Method:  http.MethodPost,
+		Uri:     a.endpoint,
+		Body:    reqJSON,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	if len(response.Cur) != 0 {
+		bidResponse.Currency = response.Cur
+	}
+
+	for _, seatBid := range response.SeatBid {
+		for i := range seatBid.Bid {
+			bid := seatBid.Bid[i]
+			bidType, err := getBidType(bid)
+			if err != nil {
+				return nil, []error{err}
```

Comment:
Returning `nil` here implies that we'd discard the rest of the elements in the `seatBid.Bid` array even if they come with valid media types. Do we want this? Or should we `continue` to process the next elements instead?
```
127             if err != nil {
128 -               return nil, []error{err}
    +               errorsArray = append(errorsArray, err)
    +               continue
129             }
adapters/smoot/smoot.go
```